core/src/main/java/hudson/model/AbstractItem.java

```diff
@@ -854,7 +854,7 @@ public void writeConfigDotXml(OutputStream os) throws IOException {
             String encoding = configFile.sniffEncoding();
             String xml = Files.readString(Util.fileToPath(configFile.getFile()), Charset.forName(encoding));
             Matcher matcher = SECRET_PATTERN.matcher(xml);
-            StringBuffer cleanXml = new StringBuffer();
+            StringBuilder cleanXml = new StringBuilder();
```

Comment:
See [this page](https://errorprone.info/bugpattern/JdkObsolete) for the reasoning.